spec/unit/util/http_proxy_spec.rb

```diff
@@ -232,6 +232,33 @@
       subject.request_with_redirects(dest, :get, 0)
     end
 
+    it 'add extra headers specified in http_extra_headers' do
+      Puppet[:http_extra_headers] = ["foo:bar", "baz:qux"]
+
+      Net::HTTP.any_instance.stubs(:head).returns(http_ok)
+      Net::HTTP.any_instance.expects(:get).with do |_, headers|
+        expect(headers)
+          .to include({'foo' => 'bar',
+                     'baz' => 'qux'})
+      end.returns(http_ok)
+
+      subject.request_with_redirects(dest, :get, 0)
+    end
+
+    it "doesn't change headers already specified" do
+      Puppet[:http_extra_headers] = ["User-Agent:foo"]
+
+      Net::HTTP.any_instance.stubs(:head).returns(http_ok)
+      Net::HTTP.any_instance.expects(:get).with do |_, headers|
+        expect(headers)
+          .to include({'Accept' => '*/*',
```

Comment:
Instead of
`expect(headers) .to include(...`
how about just checking for
`'User-Agent' => /Puppet/})`
This way if any defaults change in the future, it won't break this test.